api/http/handler/customtemplates/customtemplate_create.go

```diff
@@ -129,9 +130,20 @@ func (payload *customTemplateFromFileContentPayload) Validate(r *http.Request) e
 	if payload.Type != portainer.KubernetesStack && payload.Type != portainer.DockerSwarmStack && payload.Type != portainer.DockerComposeStack {
 		return errors.New("Invalid custom template type")
 	}
+	if !isValidNote(payload.Note) {
+		return errors.New("Invalid note. <img> tag is not supported")
+	}
 	return nil
 }
 
+func isValidNote(note string) bool {
```

Comment:
We could do both. Sanitizing the data as they are stored (including from API), is more secured. It will avoid the need for different UI/clients to sanitize them and potentially send non sanitized data to third party interfaces.

Reply:
I agree with doing both steps, I could leave this logic in the API, and also remove <img> tags from already saved templates before rendering themm.

Regarding "This function will block the update of the already saved templates that previously contains ", wouldn't that be OK, given the scenario where a user would whant to update an "unsanitized" template?

Reply:
In the scenario of the external party using the data, sanitizing at the backend is safer. Although I feel that we need to do more than just regex checking the img tag including: 

- removing dangerous tags using a library that is designed for the job (see [OWASP suggestion](https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#rule-6-sanitize-html-markup-with-a-library-designed-for-the-job))
- sanitizing other dangerous tags such as `<script>` if the input is supposed to be displayed as HTML
- encoding the received input if they are only supposed to be text
- sanitizing the other API endpoints (e.g. create/update team name)
- potential data migration

I suggest the whole sanitization in the backend can be done separately because it's a different scenario and a bigger job of its own. Keeping the `<img>` checking is OK, but the UX and implementation are likely to be changed in the above steps.

> Regarding "This function will block the update of the already saved templates that previously contains ", wouldn't that be OK, given the scenario where a user would whant to update an "unsanitized" template?

For a full backend sanitization, I was thinking about a non-blocking UX (we still sanitize the data) because we might not be able to tell the user all the exact issues when we sanitizing from the backend depending on the sanitization library we choose. However, if you choose to keep the img tag backend checking as is then this blocking is still required.

Reply:
The displaying part is ok, removing the `img` tag validation is optional so I will pass the technical review.